supportedBackends/sfs/src/main/scala/cromwell/backend/sfs/SharedFileSystemJobExecutionActor.scala

```diff
@@ -0,0 +1,43 @@
+package cromwell.backend.sfs
+
+import akka.actor.{ActorRef, Props}
+import cromwell.backend.BackendJobExecutionActor.{AbortedResponse, BackendJobExecutionResponse}
+import cromwell.backend.BackendLifecycleActor.AbortJobCommand
+import cromwell.backend.async.AsyncBackendJobExecutionActor.Execute
+import cromwell.backend.{BackendConfigurationDescriptor, BackendJobDescriptor, BackendJobExecutionActor}
+
+import scala.concurrent.{Future, Promise}
+
+class SharedFileSystemJobExecutionActor(override val jobDescriptor: BackendJobDescriptor,
+                                        override val configurationDescriptor: BackendConfigurationDescriptor,
+                                        asyncPropsCreator: Promise[BackendJobExecutionResponse] => Props)
+  extends BackendJobExecutionActor {
+
+    context.become(sharedStartup orElse super.receive)
+
+    def sharedStartup: Receive = {
+      case AbortJobCommand =>
+        context.parent ! AbortedResponse(jobDescriptor.key)
+        context.stop(self)
+      case abortResponse: AbortedResponse =>
+        context.parent ! abortResponse
+        context.stop(self)
+    }
+
+    def sharedRunning(executor: ActorRef): Receive = {
+      case AbortJobCommand =>
+        executor ! AbortJobCommand
+      case abortResponse: AbortedResponse =>
+        context.parent ! abortResponse
+        context.stop(self)
+    }
+
+    override def execute: Future[BackendJobExecutionResponse] = {
+      // Still not sure why the AsyncJobExecutionActor doesn't wait for an Akka message instead of using Scala promises
```

Comment:
AsyncBackendJobExecutionActor (ABJEA) (I missed "Backend" in my comment) uses a `completionPromise: Promise` to tell its parent actor, the BackendJobExecutionActor (BJEA), when the async job is done.

I doesn't seem obvious why we're passing around references to `scala.concurrent.Promise`s when akka has a perfectly good message system already.

Worst case, if one needed two different states and didn't want the framework to use an FSM, the BJEA could `context.become(initializing)`, start the async job, then `context.become(asyncRunning)`, and wait for the message from the ABJEA.

I didn't go down this path because the ABJEA requires the `Promise` for messaging, and I didn't want to start modifying JES, et al.